fish-rust/src/trace.rs

```diff
@@ -0,0 +1,79 @@
+use crate::{
+    common::{escape_string, EscapeFlags},
+    ffi::{self, parser_t, wcharz_t},
+    wchar::{self, wstr, L},
+    wchar_ffi::WCharToFFI,
+};
+use std::sync::atomic;
+
+#[cxx::bridge]
+mod trace_ffi {
+    extern "C++" {
+        include!("wutil.h");
+        include!("parser.h");
+        type wcharz_t = super::wcharz_t;
+        type parser_t = super::parser_t;
+    }
+
+    extern "Rust" {
+        #[cxx_name = "trace_set_enabled"]
+        fn trace_set_enabled_ffi(do_enable: bool);
+        #[cxx_name = "trace_enabled"]
+        fn trace_enabled_ffi(parser: &parser_t) -> bool;
+        #[cxx_name = "trace_argv"]
+        fn trace_argv_ffi(parser: &parser_t, command: wcharz_t, args: &Vec<wcharz_t>);
+        #[cxx_name = "trace_if_enabled"]
+        fn trace_if_enabled_ffi(parser: &parser_t, command: wcharz_t);
+    }
+}
+
+static DO_TRACE: atomic::AtomicBool = atomic::AtomicBool::new(false);
+
+fn trace_set_enabled_ffi(do_enable: bool) {
+    DO_TRACE.store(do_enable, atomic::Ordering::Relaxed);
+}
+
+fn trace_enabled_ffi(parser: &parser_t) -> bool {
```

Comment:
I think there is no reason to add the `_ffi` suffix here because
I don't think we'll ever need two versions of this function.
Today all callers pass `parser_t` no matter if C++ or Rust.
Once we port `parser_t` to `Parser` we'll switch over all callers, including the C++ ones.

Same for `trace_set_enabled`

Reply:
Applied for both